ModelBase/src/persistence/NodeIdMap.cpp

```diff
@@ -28,10 +28,17 @@
 
 namespace Model {
 
-NodeIdType NodeIdMap::id(const Node* node)
+bool NodeIdMap::containsNode(const Node* node)
 {
 	QHash<const Node*, NodeIdType>::const_iterator iter{nodeToId.find(node)};
-	if ( iter != nodeToId.end() ) return *iter;
+	if ( iter != nodeToId.end() ) return true;
+	return false;
+}
+
+NodeIdType NodeIdMap::id(const Node* node)
```

Comment:
Why did you change the implementation of `id`? Is it to reduce code duplication? That's what you should do in general, but only if the behavior stays the same. In your implementation of `id` we perform the search twice unnecessarily: once in the `containsNode` method and once again by calling `find`.

Please restore the old implementation.